### PR TITLE
Allow to configure for full classpath scan.

### PR DESCRIPTION
**Pull request by Chewbizz**
Clients can configure the extension in order for the annotated class
search process to scan the full classpath.

### CODE REVIEW

**Comment by ingwarsw** [comment]
@Chewbizz Really cool..
Thanks for PR..
Could you add any tests for that use case?
Im not sure how hard it will be but please try ..

**Comment by Chewbizz** [comment]
Sure.
I'll give it a try.

File: src/main/java/org/eu/ingwar/tools/arquillian/extension/suite/DeploymentClassFinder.java [hunk 1]
```diff
@@ -77,6 +84,25 @@
         log.log(Level.INFO,"arquillian-suite-deployment: Found class annotated with @ArquillianSuiteDeployment: {0}", type.getName());
         return type;
     }
+    
+    /**
+     * Returns name of class witch should be used as global deployment.
```
Comment (ingwarsw) [inline src/main/java/org/eu/ingwar/tools/arquillian/extension/suite/DeploymentClassFinder.java:89]: Description of that method is wrong..
Probably copy/paste

File: src/main/java/org/eu/ingwar/tools/arquillian/extension/suite/DeploymentClassFinder.java [hunk 2]
```diff
@@ -77,6 +84,25 @@
...
+     * Extension name used: suite
+     * Key: deploymentClass
+     *
+     * @param descriptor ArquillianDescriptor
+     * @return full class name from arquillian.xml
+     */
+    private static boolean isUseFullClassPathFromXml(ArquillianDescriptor descriptor) {
```
Comment (ingwarsw) [inline src/main/java/org/eu/ingwar/tools/arquillian/extension/suite/DeploymentClassFinder.java:96]: Maybe just name it `shouldScanFullClasspath`

**Comment by Chewbizz** [comment]
Tests were working using plain JUnit, but not through maven surefire, because the classloader then works differently.
See here https://maven.apache.org/surefire/maven-surefire-plugin/examples/class-loading.html
I found a workaround for surefire that can be also useful in some other setups.

It took me some time, but I learnt something :)

**Comment by ingwarsw** [comment]
@Chewbizz Thanks a lot..

**Comment by Chewbizz** [comment]
You're welcome, thank you for taking the time to review and merge :)

**Comment by ingwarsw** [comment]
@Chewbizz New version released `1.2.1` should show soon on maven central

**Pull request merged**